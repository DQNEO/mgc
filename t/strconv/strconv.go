package main

import (
	"fmt"
	"strconv"
)

func f1() {
	var a string = "10485760"
	var i int
	i, _ = strconv.Atoi(a)
	fmtPrintf(S("%d\n"), i-10485760) // 0

	a = "1"
	i, _ = strconv.Atoi(a)
	fmtPrintf(S("%d\n"), i) // 1
}

func f2() {
	var s string
	s = strconv.Itoa(0)
	fmtPrintf(S("%s\n"), s)

	s = strconv.Itoa(7)
	fmtPrintf(S("%s\n"), s)

	s = strconv.Itoa(10)
	fmtPrintf(S("%s\n"), s)

	s = strconv.Itoa(100)
	fmtPrintf(S("%s\n"), s)

	s = strconv.Itoa(1234567890)
	fmtPrintf(S("%s\n"), s)

	s = strconv.Itoa(-1)
	fmtPrintf(S("%s\n"), s)

	s = strconv.Itoa(-7)
	fmtPrintf(S("%s\n"), s)

	s = strconv.Itoa(-10)
	fmtPrintf(S("%s\n"), s)

	s = strconv.Itoa(-100)
	fmtPrintf(S("%s\n"), s)

	s = strconv.Itoa(-1234567890)
	fmtPrintf(S("%s\n"), s)

}

func main() {
	f1()
	f2()
}
