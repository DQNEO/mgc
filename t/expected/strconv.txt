0
1
0
7
10
100
1234567890
-1
-7
-10
-100
-1234567890
